profiling-ffi/src/profiles.rs

```diff
@@ -719,42 +709,38 @@ unsafe fn add_upscaling_rule(
     )
 }
 
-#[repr(C)]
-pub struct EncodedProfile {
-    start: Timespec,
-    end: Timespec,
-    buffer: ddcommon_ffi::Vec<u8>,
-    endpoints_stats: Box<ProfiledEndpointsStats>,
-}
-
 /// # Safety
 /// Only pass a reference to a valid `ddog_prof_EncodedProfile`, or null. A
 /// valid reference also means that it hasn't already been dropped (do not
 /// call this twice on the same object).
 #[no_mangle]
-pub unsafe extern "C" fn ddog_prof_EncodedProfile_drop(profile: Option<&mut EncodedProfile>) {
-    if let Some(reference) = profile {
-        // Safety: EncodedProfile's are repr(C), and not box allocated. If the
-        // user has followed the safety requirements of this function, then
-        // this is safe.
-        std::ptr::drop_in_place(reference as *mut _)
+pub unsafe extern "C" fn ddog_prof_EncodedProfile_drop(
+    profile: *mut Handle<internal::EncodedProfile>,
+) {
+    // Technically, this function has been designed so if it's double-dropped
+    // then it's okay, but it's not something that should be relied on.
+    if !profile.is_null() {
+        drop((*profile).take())
     }
 }
 
-impl From<internal::EncodedProfile> for EncodedProfile {
-    fn from(value: internal::EncodedProfile) -> Self {
-        let start = value.start.into();
-        let end = value.end.into();
-        let buffer = value.buffer.into();
-        let endpoints_stats = Box::new(value.endpoints_stats);
-
-        Self {
-            start,
-            end,
-            buffer,
-            endpoints_stats,
-        }
-    }
+/// Given an EncodedProfile, get a slice representing the bytes in the pprof.
+/// This slice is valid for use until the encoded_profile is modified in any way (e.g. dropped or
+/// consumed).
+/// # Safety
+/// Only pass a reference to a valid `ddog_prof_EncodedProfile`.
+#[no_mangle]
+#[must_use]
+#[named]
+pub unsafe extern "C" fn ddog_prof_EncodedProfile_get_pprof_bytes<'a>(
```

Comment:
Minor: Slightly shorter name, and doesn't promise anything about the format ;)

```suggestion
pub unsafe extern "C" fn ddog_prof_EncodedProfile_bytes<'a>(
```